### DISCUSSION TITLE
SQL Backend

### DISCUSSION THREAD

**Discussion by tastyfrankfurt** [SigmaHQ/pySigma#131] [General]
Hey Guys,

You have done a awesome job and putting this together to support the CyberSec community.

There are alot of dataware house tools that utilise SQL as a query language, I am happy to have a crack at the backend and was wondering if you can create a project for it for me to contribute against.  I have already started with a cookiecutter and have been working through logically breaking things up but there are some limitations that are causing me some headaches when producing the query. 

One example is there a way to just access the fields rather than just one big query that gets returned by the backend?

Thanks

Matt

**Comment by thomaspatzke** [comment]
> You have done a awesome job and putting this together to support the CyberSec community.

Thanks 😊

> There are alot of dataware house tools that utilise SQL as a query language, I am happy to have a crack at the backend and was wondering if you can create a project for it for me to contribute against.

You mean a placeholder project you can push your PR to? Below the SigmaHQ organization? `pySigma-backend-sql`?

> I have already started with a cookiecutter and have been working through logically breaking things up but there are some limitations that are causing me some headaches when producing the query.
> 
> One example is there a way to just access the fields rather than just one big query that gets returned by the backend?

All fields within a Sigma rule? It requires to iterate recursively over detections. If I remember correctly there was already someone who asked for it, so I could implement it into the pySigma core.

**Reply by Mrlogisim** [reply]
Hello, I have a question about data warehouses.

I don’t know much about data warehouses, and now I have to work on a project based on a banking scenario where a manager wants to analyze transaction movements.

I would like to know: where is a data warehouse actually stored? For example, if I create a PostgreSQL database that includes the data marts of a data warehouse, would that be considered the data warehouse?

For instance, if I create a database called "dw_bank", where I store information from a transaction data mart and later add information from other data marts into the same database, would that be considered the data warehouse?

**Reply by thomaspatzke** [reply]
Hi! Sorry, but I think this is totally offtopic here. This project is about security detections, not data warehouses.

**Reply by Mrlogisim** [reply]
Ahh, no problem. Sorry for the inconvenience.
